sdk/src/logs/logger.cc

```diff
@@ -48,44 +49,64 @@ const opentelemetry::nostd::string_view Logger::GetName() noexcept
 
 opentelemetry::nostd::unique_ptr<opentelemetry::logs::LogRecord> Logger::CreateLogRecord() noexcept
 {
-  auto recordable = context_->GetProcessor().MakeRecordable();
+  std::unique_ptr<Recordable> recordable;
+  try
+  {
+    recordable = context_->GetProcessor().MakeRecordable();
+  }
+  catch (const std::exception &e)
+  {
+    OTEL_INTERNAL_LOG_ERROR("[Logger::CreateLogRecord]"
+                            << "Error while creating recordable: " << e.what()
+                            << " Unable to create new Log Record");
+    return nullptr;
```

Comment:
Returning a nullptr is likely to break things.

Please revert all changes done to this method, this needs to be addressed separately.

Reply:
Reverting. We'll take this in next PR